### PR TITLE
Add `log_weight_decay` argument to `LearningRateMonitor` callback

### LINKED ISSUE #18117: Add `log_weight_decay` option to `LearningRateMonitor` callback
### Description & Motivation

In more sophisticated training settings, we usually apply a dedicated weight decay scheduler (for example DINO). Thus, it will be great to monitor these statistics, optionally, from the `LearningRateMonitor` callback.

### Pitch

A solution should look something like the following, but build in the LearningRateMonitor:
```py
"""Learning rate monitor logger callback."""
from typing import Any, Dict, List

from pytorch_lightning import callbacks
from torch import optim


class LearningRateMonitor(callbacks.LearningRateMonitor):
    """Extends the PL version with the ability to log the weight decay."""

    def __init__(
        self,
        logging_interval: str | None = None,
        log_momentum: bool = False,
        log_weight_decay: bool = False,
    ) -> None:
        """Initializes the callback."""
        super().__init__(logging_interval, log_momentum)
        self.log_weight_decay = log_weight_decay

    def _get_lr_momentum_stat(
        self, optimizer: optim.Optimizer, names: List[str]
    ) -> Dict[str, float]:
        """Gathers and returns the learning rate, momentum and weight decay statistics."""
        lr_momentum_wd_stat = {}
        param_groups = optimizer.param_groups
        use_betas = "betas" in optimizer.defaults

        for pg, name in zip(param_groups, names):
            lr = self._extract_lr(pg, name)
            lr_momentum_wd_stat.update(lr)
            momentum = self._extract_momentum(
                param_group=pg, name=name.replace(name, f"{name}-momentum"), use_betas=use_betas
            )
            lr_momentum_wd_stat.update(momentum)
            weight_decay = self._extract_weight_decay(pg, name + "-weight_decay")
            lr_momentum_wd_stat.update(weight_decay)

        return lr_momentum_wd_stat

    def _extract_weight_decay(self, param_group: Dict[str, Any], name: str) -> Dict[str, Any]:
        """Extracts the weight decay statistics from a parameter group."""
        if not self.log_weight_decay:
            return {}

        wd = param_group["weight_decay"]
        return {name: wd}
```

### Alternatives

_No response_

### Additional context

_No response_

cc @borda

### PR DESCRIPTION
**Pull request by ioangatop**
## What does this PR do?

Fixes #18117

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

File: src/lightning/pytorch/callbacks/lr_monitor.py [hunk 1]
```diff
@@ -147,6 +154,7 @@ def _check_no_key(key: str) -> bool:
         names_flatten = list(itertools.chain.from_iterable(names))
         self.lrs = {name: [] for name in names_flatten}
         self.last_momentum_values = {name + "-momentum": None for name in names_flatten}
+        self.last_weight_decay_values = {name + "-weight_decay": None for name in names_flatten}
```
[the remark this answers is not in the document]
Comment (ioangatop) [reply src/lightning/pytorch/callbacks/lr_monitor.py:157]: 100% - done!

File: src/lightning/pytorch/callbacks/lr_monitor.py [hunk 2]
```diff
@@ -202,20 +212,22 @@ def _extract_stats(self, trainer: "pl.Trainer", interval: str) -> Dict[str, floa
...
         for pg, name in zip(param_groups, names):
             lr = self._extract_lr(pg, name)
-            lr_momentum_stat.update(lr)
+            stats.update(lr)
             momentum = self._extract_momentum(
                 param_group=pg, name=name.replace(name, f"{name}-momentum"), use_betas=use_betas
             )
-            lr_momentum_stat.update(momentum)
+            stats.update(momentum)
+            weight_decay = self._extract_weight_decay(pg, name + "-weight_decay")
```
Comment (carmocca) [inline src/lightning/pytorch/callbacks/lr_monitor.py:227]: nit: we use fstring directly above

```suggestion
            weight_decay = self._extract_weight_decay(pg, f"{name}-weight_decay")
```

**Review by carmocca — APPROVED** [review]
LGTM
